verifier/src/main/java/com/aws/jverify/verifier/JavaToDafnyCompiler.java

```
@@ -0,0 +1,485 @@
+                    } else {
+                        returnName = new Name(origin, "r");
+                    }
+                    var f = new Formal(toOrigin(method.getReturnType()), returnName, returnType,
+                            false, false, null, null, false, false, false, null);
+                    outs.add(f);
+                }
+
+                if (annotationsByName.containsKey("Proof")) {
+                    return new Method(origin, name, null, isStatic, false, null, List.of(),
+                            ins, req, ens, new Specification<FrameExpression>(origin, List.of(), null),
+                            new Specification<>(origin, List.of(), null), outs,
+                            new Specification<FrameExpression>(origin, List.of(), null),
+                            new BlockStmt(origin, null, statements), false);
+                } else {
+                    return new Method(origin, name, null, isStatic, false, null, List.of(),
+                            ins, req, ens, new Specification<FrameExpression>(origin, List.of(), null),
+                            new Specification<>(origin, List.of(), null), outs,
+                            new Specification<FrameExpression>(origin, List.of(), null),
+                            new BlockStmt(origin, null, statements), false);
+                }
+            }
+        }
+        throw new NotImplementedException(member.getClass().getName());
+    }
+
+    private List<Statement> processStatementWithHeader(JCTree.JCStatement statement,
+                                                        Function<JCTree.JCStatement, Boolean> checkPossibleHeader) {
+        var statements = statement instanceof JCTree.JCBlock block ? block.stats : List.of(statement);
+        return processStatementsWithHeader(statements, checkPossibleHeader);
+    }
+    
+    private List<Statement> processStatementsWithHeader(List<JCTree.JCStatement> stats, 
```

It looks like if "header" statements appear after the first non-header statement, they'll be silently ignored by `translateStatement` - we should consider emitting a warning to avoid user confusion.